src/structure/grid.ts

```diff
@@ -120,11 +128,11 @@ export class ViewersGrid {
     /// List of structures in the dataset
     private _structures: Structure[] | UserStructure[];
     /// List of properties in the dataset
-    private _properties: { [name: string]: Property };
+    private _properties: Record<string, Property>;
```

Comment:
Since you already filter out the string and structure properties

```suggestion
    private _properties: Record<string, number[]>;
```

and changing the comment to `List of per-atom properties in the dataset`